src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -153,24 +165,108 @@ void GcsPlacementGroupManager::HandleCreatePlacementGroup(
     ray::rpc::SendReplyCallback send_reply_callback) {
   auto placement_group_id =
       PlacementGroupID::FromBinary(request.placement_group_spec().placement_group_id());
-  const auto &name = request.placement_group_spec().name();
-  const auto &strategy = request.placement_group_spec().strategy();
-  RAY_LOG(INFO) << "Registering placement group, placement group id = "
-                << placement_group_id << ", name = " << name
-                << ", strategy = " << PlacementStrategy_Name(strategy);
   auto placement_group = std::make_shared<GcsPlacementGroup>(request);
+
+  RAY_LOG(INFO) << "Registering placement group, " << placement_group->DebugString();
+  // We need this call here because otherwise, if placement group is removed right after
+  // here, it can cause inconsistent states.
+  registered_placement_groups_.emplace(placement_group_id, placement_group);
+
   RAY_CHECK_OK(gcs_table_storage_->PlacementGroupTable().Put(
       placement_group_id, placement_group->GetPlacementGroupTableData(),
-      [this, request, reply, send_reply_callback, placement_group_id, name,
-       strategy](Status status) {
+      [this, request, reply, send_reply_callback, placement_group_id,
+       placement_group](Status status) {
         RAY_CHECK_OK(status);
-        RegisterPlacementGroup(request, [reply, send_reply_callback, placement_group_id,
-                                         name, strategy]() {
-          RAY_LOG(INFO) << "Finished registering placement group, placement group id = "
-                        << placement_group_id << ", name = " << name
-                        << ", strategy = " << strategy;
-          GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::OK());
-        });
+        if (registered_placement_groups_.find(placement_group_id) ==
+            registered_placement_groups_.end()) {
+          std::stringstream stream;
+          stream << "Placement group of id " << placement_group_id
+                 << " has been removed before registration.";
+          GCS_RPC_SEND_REPLY(send_reply_callback, reply, Status::NotFound(stream.str()));
+          return;
+        }
+
+        RegisterPlacementGroup(
+            request, [reply, send_reply_callback, placement_group](Status status) {
+              if (status.ok()) {
+                RAY_LOG(INFO) << "Finished registering placement group, "
+                              << placement_group->DebugString();
+              } else {
+                RAY_LOG(WARNING) << "Failed to register placement group, "
+                                 << placement_group->DebugString();
```

Comment:
If you do keep the logging, make sure to include the error message in the logging.